src/gravitational_waves/gw.cpp

```diff
@@ -220,61 +330,412 @@ Nintegrate_SNR(GravitationalWave &obj, const double fmin, const double fmax)
+  std::vector<double> LeptonSq_False =
+      BACalc.modelPointer->LeptonMassesSquared(FalseVacuum);
+  std::vector<double> QuarkSq_False =
+      BACalc.modelPointer->QuarkMassesSquared(FalseVacuum);
+  std::vector<double> GaugeSq_False =
+      BACalc.modelPointer->GaugeMassesSquared(FalseVacuum, Tstar);
+
+  std::vector<double> HiggsSq_True =
+      BACalc.modelPointer->HiggsMassesSquared(TrueVacuum, Tstar);
+  std::vector<double> LeptonSq_True =
+      BACalc.modelPointer->LeptonMassesSquared(TrueVacuum);
+  std::vector<double> QuarkSq_True =
+      BACalc.modelPointer->QuarkMassesSquared(TrueVacuum);
+  std::vector<double> GaugeSq_True =
+      BACalc.modelPointer->GaugeMassesSquared(TrueVacuum, Tstar);
+
+  double P_LO = 0; // Pressure at LO. 1 -> 1
+
+  for (auto massSq : HiggsSq_True)
+    P_LO += massSq;
+  for (auto massSq : GaugeSq_True)
+    P_LO += 3 * massSq; // 3 from polarization
+  for (auto massSq : LeptonSq_True)
+    P_LO += massSq / 2.;
+  for (auto massSq : QuarkSq_True)
+    P_LO += 3 * massSq / 2.; // 3 from colour
+  for (auto massSq : HiggsSq_False)
+    P_LO -= massSq;
+  for (auto massSq : GaugeSq_False)
+    P_LO -= 3 * massSq; // 3 from polarization
+  for (auto massSq : LeptonSq_False)
+    P_LO -= massSq / 2.;
+  for (auto massSq : QuarkSq_False)
+    P_LO -= 3 * massSq / 2.; // 3 from colour
+
+  P_LO *= pow(Tstar, 2) / 24.; // pressure LO normalization
+
+  // Pressure at NLO. 1 -> N
+  // This only works for the SU(2)_L x U(1)_Y gauge group. But can be easily
+  // generalized.
+  const double W_coupling = BACalc.modelPointer->SMConstants.C_g / sqrt(2);
+  const double Z_coupling =
+      BACalc.modelPointer->SMConstants.C_g /
+      sqrt(1. - BACalc.modelPointer->SMConstants.C_sinsquaredWeinberg);
+  const double A_coupling =
+      BACalc.modelPointer->SMConstants.C_g *
+      sqrt(BACalc.modelPointer->SMConstants.C_sinsquaredWeinberg);
+  double P_NLO = 0;
+  if (pnlo_scaling == 1)
+  {
+    // Mass matrix is ordered
+    P_NLO += sqrt(GaugeSq_True.at(0)) * pow(A_coupling, 2);
+    P_NLO += 2 * sqrt(GaugeSq_True.at(1)) * pow(W_coupling, 2);
+    P_NLO += sqrt(GaugeSq_True.at(3)) * pow(Z_coupling, 2);
+
+    P_NLO -= sqrt(GaugeSq_False.at(0)) * pow(A_coupling, 2);
+    P_NLO -= 2 * sqrt(GaugeSq_False.at(1)) * pow(W_coupling, 2);
+    P_NLO -= sqrt(GaugeSq_False.at(3)) * pow(Z_coupling, 2);
+
+    P_NLO *= pow(Tstar, 3);
+  }
+  else if (pnlo_scaling == 2)
+  {
+    P_NLO = 2 * pow(W_coupling, 2) + pow(Z_coupling, 2) + pow(A_coupling, 2);
+    P_NLO *= pow(Tstar, 4); // pressure NLO normalization
+  }
+  const double alpha_infty    = P_LO / RhoGamma;
+  const double gamma_run_away = Rstar / (3. * R0);
+  if (BACalc.GetPTStrength() - alpha_infty < 0)
+    return 0.; // Not enough pressure to drive the bubble wall.
+  const double gamma_eq = pow((dV - P_LO) / P_NLO, 1. / pnlo_scaling);
+  if (gamma_eq < 1) return 0.; // dV - P_LO < P_NLO. Unphysical gamma < 1;
```

Comment:
What is with the commented code?
```suggestion
  if (gamma_eq < 1) return 0.; 
```

Reply:
yes, this sounds better

Reply:
[Done.](https://github.com/phbasler/BSMPT/pull/180/commits/6e023a438c923d346fb7106393dc9d6b78b8026c)